Assets/Scripts/Controllers/HistoryController.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Controllers;
+using Services.History;
+
+namespace Controllers
+{
+    public class HistoryController: MonoBehaviour
+    {
+        private Stack<ICommand> _Undocommands = new Stack<ICommand>();
+        private Stack<ICommand> _Redocommands = new Stack<ICommand>();
```

Comment:
- Inconsistent naming
- Use readonly whenever you can
- Use more meaningful names
```suggestion
        private readonly Stack<ICommand> redoActionStack = new Stack<ICommand>();
```